### ISSUE TITLE
Abstract class method cannot be called, but only when init_subclass defined

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#5159]
```python
from abc import abstractmethod

class Parametrization:
    def __init_subclass__(cls) -> None:  # Removing this method makes the error disappear!
        super().__init_subclass__()

    @classmethod
    @abstractmethod
    def f(cls) -> int:
        raise NotImplementedError

class Samplable(Parametrization):
    pass

def g(s: type[Samplable]):
    s.f()  # error: Method "f" cannot be called because it is abstract (reportGeneralTypeIssues)
```

**Comment by erictraut** [comment]
Yes, this is a false positive. It should not generate an error (regardless of whether `__init_subclass__` is present).

This will be fixed in the next release.

**Comment by NeilGirdhar** [comment]
Perfect, thank you!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.310, which I just published. It will also be included in a future release of pylance.
